cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -0,0 +1,634 @@
+                .build();
+    }
+
+    private MetadataStore.SegmentInfo resetSegmentID(MetadataStore.SegmentInfo segmentInfo) {
+        MetadataStore.SegmentInfo resetSegmentInfo = MetadataStore.SegmentInfo.builder()
+                .segmentId(NO_STREAM_SEGMENT_ID)
+                .properties(segmentInfo.getProperties())
+                .build();
+        return resetSegmentInfo;
+    }
+
+    /**
+     * Writes all the parsed TableSegment operations from raw chunks into storage_metadata of the debug
+     * segment container.
+     *
+     * @param container Container whose storage_metadata table segment is populated.
+     * @param tableSegment name of container metadata segment.
+     * @param tableSegmentOperations Operations parsed from storage metadata raw chunk file bytes for container metadata.
+     * @throws Exception
+     */
+    private void writeEntriesToStorageMetadata(DebugStreamSegmentContainer container, String tableSegment, List<TableSegmentUtils.TableSegmentOperation> tableSegmentOperations) throws Exception {
+        output("Writing entries to storage metadata");
+        ContainerTableExtension tableExtension = container.getExtension(ContainerTableExtension.class);
+        for (TableSegmentUtils.TableSegmentOperation operation : tableSegmentOperations) {
+            TableSegmentEntry entry = operation.getContents();
+            TableEntry unversionedEntry = TableEntry.unversioned(new ByteBufWrapper(entry.getKey().getKey()), new ByteBufWrapper(entry.getValue()));
+            String segment = new String(unversionedEntry.getKey().getKey().getCopy());
+            if (!allowSegment(segment)) {
+                continue;
+            }
+            if (operation instanceof TableSegmentUtils.PutOperation) {
+                tableExtension.put(tableSegment, Collections.singletonList(unversionedEntry), TIMEOUT).join();
+            } else {
+                tableExtension.remove(tableSegment, Collections.singletonList(unversionedEntry.getKey()), TIMEOUT);
+            }
+        }
+    }
+
+    private boolean allowSegment(String segmentName) {
+        // Not recovering/backing up below segments to allow recovery to work.
+        // If we come up with alternative ways of handling below issues we can avoid skipping these segments.
+        // 1. scaleGroup: scaleGroup segment stores ReaderGroup state. If this segment is recovered
+        //    it restores all old ReaderGroup state having readers assigned to segments we want to read from.
+        //    As a result reads get indefinitely blocked.
+        // 2. Recovering EVENT_PROCESSOR_SEGMENT leads to length mismatch issues between container and storage
+        //    metadata as EVENT_PROCESSOR_SEGMENTS are created by default when container starts and then as part
+        //    of recovery, when container comes up we update the lenghts in the metadata.
+        if (segmentName.contains("scaleGroup") || segmentName.contains(EVENT_PROCESSEOR_SEGMENT)) {
```

Comment:
If you are adding constants for all these hardcoded strings, do the same for `scaleGroup` for consistency.

Reply:
done.